src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java

```diff
@@ -376,6 +381,56 @@ void should_return_a_200_with_empty_list_when_the_user_did_not_have_any_roles()
 
     }
 
+    @DisplayName("Should return 502 when camunda service is down")
+    @Test
+    void should_return_status_code_502_when_camunda_service_is_down() throws Exception {
+
+        searchEventAndCase = new SearchEventAndCase(
+            "some-caseId",
+            "some-eventId",
+            "ia",
+            "asylum"
+        );
+        mockServices.mockServiceAPIs();
+
+        doThrow(FeignException.BadGateway.class)
+            .when(camundaServiceApi)
+            .evaluateDMN(any(), any(), any(), any());
+
+        when(launchDarklyFeatureFlagProvider.getBooleanValue(any(), any(), any())).thenReturn(true);
+
+        when(allowedJurisdictionConfiguration.getAllowedJurisdictions())
+            .thenReturn(List.of("wa", "ia", "sscs", "civil"));
+
+        when(allowedJurisdictionConfiguration.getAllowedCaseTypes())
+            .thenReturn(List.of("asylum", "wacasetype", "sscs", "civil"));
+
+        when(authTokenGenerator.generate())
+            .thenReturn(IDAM_AUTHORIZATION_TOKEN);
+
+        mockMvc.perform(
+                post("/task/search-for-completable")
+                    .header(AUTHORIZATION, IDAM_AUTHORIZATION_TOKEN)
+                    .header(SERVICE_AUTHORIZATION, SERVICE_AUTHORIZATION_TOKEN)
+                    .content(asJsonString(searchEventAndCase))
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+            )
+            .andExpectAll(
+                status().is5xxServerError(),
```

Comment:
should check the status is 502?

Reply:
This method checks error code starts with 5 or not.
Here is their explanation 'Assert the response status code is in the 5xx range'
Should I take any action?

Reply:
as previous comment

Reply:
in line 424 also check it 
jsonPath("$.status").value(502),

Reply:
No this is different.  That's in the body more for debugging convenience.  The overall http status is not in the body.